### PR TITLE
gigadevice: multiple minor fixes/cleanups

### PR DESCRIPTION
**Pull request by gmarull**
This PR includes a few minor fixes/cleanups for the Gigadevice platform. Refer to commit messages for more details.

### CODE REVIEW

File: dts/arm/gigadevice/gd32f403/gd32f403.dtsi [hunk 1]
```diff
@@ -16,8 +17,8 @@
 		#size-cells = <0>;
 
 		cpu0: cpu@0 {
-			device_type = "cpu";
```
Comment (galak) [inline dts/arm/gigadevice/gd32f403/gd32f403.dtsi:19]: its ok to keep `device_type = "cpu";`

**Review by galak — DISMISSED** [review]
lgtm.  The `device_type` for cpu is fine to keep.

**Review by nandojve — DISMISSED** [review]

File: dts/arm/gigadevice/gd32f403/gd32f403.dtsi (hunk 1 above)
Comment (gmarull) [reply dts/arm/gigadevice/gd32f403/gd32f403.dtsi:19]: compatibility with IEEE 1275 is not needed here, this is why I removed it.

**Review by cameled — APPROVED** [review]
LGTM.

**Review by galak — APPROVED** [review]

**Pull request merged**